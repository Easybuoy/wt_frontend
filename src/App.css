@import "~@fullcalendar/core/main.css";
@import "~@fullcalendar/daygrid/main.css";
@import "~@fullcalendar/timegrid/main.css";

* {
  box-sizing: border-box;
}

.App {
  text-align: center;
}

.error {
  color: red;
}
