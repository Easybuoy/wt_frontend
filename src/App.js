import React from "react";
import { ThemeProvider, CSSReset, theme } from "@chakra-ui/core";
import { Route, BrowserRouter as Router } from "react-router-dom";
import "./App.css";
import Login from "./components/LogIn";

function App() {
  return (
    <ThemeProvider
      theme={{
        ...theme,
        fonts: { ...theme.fonts, body: "Roboto", heading: "Ubuntu" }
      }}
    >
      <CSSReset />
      <Router>
        <Route exact path="/" component={Login} />
      </Router>
    </ThemeProvider>
  );
}

export default App;
