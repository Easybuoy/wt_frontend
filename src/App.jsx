import React from "react";
import { Route } from "react-router-dom";
import Onboarding from "./components/auth/Onboarding";
import "./App.css";
import PrivateRoute from "./components/common/PrivateRoute";
import Login from "./components/auth/LogIn";
import Dashboard from "./components/dashboard/Dashboard";
import SignUp from "./components/auth/SignUp";
import WorkoutList from "./components/workouts/WorkoutList";
import WorkoutDetail from "./components/workouts/WorkoutDetail";
import Exercises from "./components/exercise/ExerciseList";
import ExerciseDetail from "./components/exercise/ExerciseDetail";
import Navigation from "./components/common/Navigation";
import WorkoutHistory from "./components/workouts/WorkoutHistory";
import ScheduleDetail from "./components/schedule/ScheduleDetail";
import CustomWorkoutDetail from "./components/workouts/CustomWorkoutDetail";

function App() {
  return (
    <div className="App" data-testid="App">
      <Navigation />
      <Route exact path="/login" component={Login} />
      <PrivateRoute exact path="/exercises" component={Exercises} />
      <PrivateRoute exact path="/exercise/:id" component={ExerciseDetail} />
      <PrivateRoute exact path="/onboarding" component={Onboarding} />
      <Route exact path="/signup" component={SignUp} />
      <PrivateRoute exact path="/" component={Dashboard} />
      <PrivateRoute exact path="/schedule" component={ScheduleDetail} />
      <PrivateRoute exact path="/workouts" component={WorkoutList} />
      <PrivateRoute exact path="/workout/:id" component={WorkoutDetail} />
      <PrivateRoute
        exact
        path="/my/workout/:id"
        component={CustomWorkoutDetail}
      />
      <PrivateRoute exact path="/workouthistory" component={WorkoutHistory} />
    </div>
  );
}

export default App;
