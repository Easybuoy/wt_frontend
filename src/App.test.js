import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import { BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider, theme } from "@chakra-ui/core";

const customRender = component =>
  render(
    <Router>
      <ThemeProvider
        theme={{
          ...theme,
          fonts: { ...theme.fonts, body: "Roboto", heading: "Ubuntu" }
        }}
      >
        {component}
      </ThemeProvider>
    </Router>
  );

test("renders without crashing", () => {
  const { getByTestId } = customRender(<App />);
  const app = getByTestId("App");
  expect(app).toBeInTheDocument();
});
