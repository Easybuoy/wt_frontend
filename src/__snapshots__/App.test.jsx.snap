// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the App component correctly 1`] = `
<div
  className="App"
  data-testid="App"
>
  <ErrorBoundary>
    <withRouter(Navigation) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/accountrecovery"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/messages"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/messages/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/accountrecovery/:token"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/profile"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/exercises"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/exercise/:id"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/onboarding"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/signup"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/workouts"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/schedule"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/workout/:id"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/my/workout/:id"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/workouthistory"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/buddies"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/gallery"
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
    <withRouter(Footer) />
  </ErrorBoundary>
</div>
`;
